### DISCUSSION TITLE
Frequent memory growth of the quorum queue

### DISCUSSION THREAD

**Discussion by mayee** [rabbitmq/rabbitmq-server#3209] [Other] [answered]
Hi,

Summary:
Rabbitmq 3.8.9, Erlang 23.1
Cluster of 3 nodes, memory limit up to 100G per node.

Global counts 
Connections: 6112 Channels: 10173 Exchanges: 8 Queues: 7360 Consumers: 6884

We have around 6k+ devices connecting with mqtt protocol, the memory growth rate is very fast (10g a day)
quorum_queue_procs
![rabbit](https://raw.githubusercontent.com/mayee/img-folder/master/rabbit01.png)

![rabbit](https://github.com/mayee/img-folder/blob/master/rabbit02.png?raw=true)

![rabbit](https://github.com/mayee/img-folder/blob/master/rabbit03.png?raw=true)

TTL is not supported for quorum queues. I wonder how to avoid it.

**Comment by kjnilsson** [comment]
Hi, please see: https://www.rabbitmq.com/mqtt.html#consensus

On Wed, 21 Jul 2021 at 11:48, mayee ***@***.***> wrote:

> Hi,
>
> Summary:
> Rabbitmq 3.8.9, Erlang 23.1
> Cluster of 3 nodes, memory limit up to 100G per node.
>
> Global counts
> Connections: 6112 Channels: 10173 Exchanges: 8 Queues: 7360 Consumers: 6884
>
> We have around 6k+ devices connecting with mqtt protocol, the memory
> growth rate is very fast (10g a day)
> quorum_queue_procs
> [image: rabbit]
> <https://raw.githubusercontent.com/mayee/img-folder/master/rabbit01.png>
>
> [image: rabbit]
> <https://github.com/mayee/img-folder/blob/master/rabbit02.png?raw=true>
>
> [image: rabbit]
> <https://github.com/mayee/img-folder/blob/master/rabbit03.png?raw=true>
>
> TTL is not supported for quorum queues. I wonder how to avoid it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/issues/3208>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFFYFMB6L6Y5ZRSULG3TY2QYDANCNFSM5AXWMFKA>
> .
>


-- 
*Karl Nilsson*

**Comment by michaelklishin** [comment, marked as answer]
Somehow my earlier comment has gone missing.

What you are looking at is a Raft-based feature but it has nothing to do with quorum queues. You can and in most cases,
should [configure Raft settings](https://www.rabbitmq.com/mqtt.html#consensus) very differently from defaults if you only use MQTT tracking. As of 3.9 those features
can be given separate defaults so it will not be required at some point.

Also, [`3.8.9` is 10 months old](https://www.rabbitmq.com/changelog.html). Please upgrade to `3.8.19` which will include https://github.com/rabbitmq/rabbitmq-server/pull/2692 among other things.

**Reply by mayee** [reply]
we are trying to use the latest version, thanks

**Comment by michaelklishin** [comment]
You can also [lower the amount of per-quorum queue memory used](https://www.rabbitmq.com/quorum-queues.html#resource-use).
The defaults are optimal for 8-12 queues at peak throughput but not 7-8K relatively idle ones.

**Comment by johanrhodin** [comment]
This looks like https://github.com/rabbitmq/rabbitmq-server/discussions/2688. Upgrade all cluster nodes to latest versions  of RabbitMQ and Erlang.

**Reply by kjnilsson** [reply]
yeah on closer inspection that is very likely
